typescript-codegen.ts

```diff
@@ -0,0 +1,241 @@
+            fs.appendFileSync(
+                `${generationPath}/${resource.name}.ts`,
+                `import BasePokeResource from '../../BasePokeResource';${EOL}${EOL}`
+            );
+            fs.appendFileSync(
+                `${generationPath}${resource.name}.ts`,
+                `export const ${resource.name.toUpperCase()}_ROUTE : string = "${
+                    resource.route
+                }";${EOL + EOL}`
+            );
+            resource.typescriptInterfaceContent.lines.forEach((line) => {
+                if (line.includes("export interface " + resource.name)) {
+                    line = `export class ${resource.name} extends BasePokeResource {`;
+                }
+                if (line.includes("export class Convert")) {
+                    line = `export class ${resource.name}Converter {`;
+                }
+                if (line.includes("[property: string]: any;")) {
+                    line = "";
+                    if (resource.jsonContent.operations.includes("DELETE")) {
+                        line += this.generateDelete(resource.name);
+                    }
+                    if (resource.jsonContent.operations.includes("MODIFY")) {
+                        line += this.generatePatch(resource.name);
+                    }
+                }
+                fs.appendFileSync(
+                    `${generationPath}/${resource.name}.ts`,
+                    `${line}${EOL}`
+                );
+            });
+        });
+    };
+
+    public GeneratePrimitives = async (generationPath: string) => {
+        if (this.primitives.length == 0 || this.primitivesPath == null) {
+            console.log("Primitives were not generated!")
+            return;
+        }
+        this.primitives.forEach((primitive) => {
+            console.log(primitive.name);
+            fs.writeFileSync(
+                `${generationPath}/${primitive.name}.ts`,
+                ""
+            );
+            primitive.typescriptInterfaceContent.lines.forEach((line) => {
+                fs.appendFileSync(
+                    `${generationPath}/${primitive.name}.ts`,
+                    `${line}${EOL}`
+                );
+            });
+        });
+    };
+}
+
+const workingDirectory = process.cwd();
+const primitivesSchemasPath = `${workingDirectory}/primitives`;
+const resourcesSchemasPath = `${workingDirectory}/resources`;
+const primitivesGenerationPath = `${workingDirectory}/PokeClient.ts/generated/primitives`;
+const resourcesGenerationPath = `${workingDirectory}/PokeClient.ts/generated/resources`;
+
+const codegenHelper : TypescriptCodegenHelper = new TypescriptCodegenHelper();
+codegenHelper.LoadPrimitives(primitivesSchemasPath).then((val) => {
+    codegenHelper.GeneratePrimitives(primitivesGenerationPath);
+});
+codegenHelper.LoadResources(resourcesSchemasPath).then((val) => {
+    codegenHelper.GenerateResources(resourcesGenerationPath);
+});
```

Comment:
This goes concurrently.... whyyy???

Reply:
Wdym? You wanna sync the Loads/GenerateResources in some way?